daemon/api.go

```diff
@@ -434,10 +434,11 @@ func storeUpdates(c *Command, r *http.Request, user *auth.UserState) Response {
 		})
 	}
 
-	var auther store.Authenticator
-	if user != nil {
-		auther = user.Authenticator()
+	auther, err := c.d.auther(r)
```

Comment:
We've just been refactoring the whole code in the exact opposite direction, by taking away the re-parsing of the request and considering the user that we verify once in the entrance of the request. I'd prefer to not undo this, and this seems doable.

It would be nice to have an interactive conversation to sort this out.